### PR TITLE
[ADVAPP-979]: Enhance the user list experience by introducing a filter for "no assigned team"

### PR DESCRIPTION
**Pull request by amit-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-979

### Technical Description

> Enhance the user list experience by introducing a filter for "no assigned team"

### Any deployment steps required?

>No

### Are any Feature Flags and/or Data Migrations that can eventually be removed Added?

> No

_______________________________________________

### CODE REVIEW

File: app-modules/authorization/tests/Feature/Filament/Resources/User/ListUsersTest.php [hunk 1]
```diff
@@ -264,3 +264,36 @@
...
+            $userInTeamA
+        )
+        ->filterTable('teams', [$teamB->id])
+        ->assertCanSeeTableRecords(
+            $userInTeamB
+        )
+        ->filterTable('teams', [])
+        ->assertCanSeeTableRecords(
+            $unassignedUser
+        );
```
[the remark this answers is not in the document]
Comment (amit-canyon) [reply app-modules/authorization/tests/Feature/Filament/Resources/User/ListUsersTest.php:298]: @danharrin  & @Orrison 
Sure , i will this. Also i will make same change my another PR that related to the filter ass well. 
Thanks!.

**Review by danharrin — COMMENTED** [review]
The options are not grouped yet, did you try that bit of the ticket?

**Comment by amit-canyon** [comment]
> The options are not grouped yet, did you try that bit of the ticket?

@danharrin  Sorry, but i didn't getting you, please explain little bit more here. Thanks.

File: app/Filament/Resources/UserResource/Pages/ListUsers.php [hunk 2]
```diff
@@ -150,14 +152,26 @@ public function table(Table $table): Table
...
             ->defaultSort('name', 'asc');
     }
 
+    public function getTeamsOption(): array
+    {
+        $teams = Team::query()->take(50)->pluck('name', 'id')->toArray();
+
+        return [
+            'unassigned' => 'Unassigned',
+            ...$teams,
```
Comment (Orrison) [inline app/Filament/Resources/UserResource/Pages/ListUsers.php:171]: It is expected that this options are grouped: https://filamentphp.com/docs/3.x/forms/fields/select#grouping-options

**Review by Orrison — CHANGES_REQUESTED** [review]

**Review by Orrison — APPROVED** [review]

**Pull request merged**